test/integration/apiserver/admissionwebhook/admission_test.go

```diff
@@ -99,6 +99,8 @@ var (
 	customTestFuncs = map[schema.GroupVersionResource]map[string]testFunc{
 		gvr("", "v1", "namespaces"): {"delete": testNamespaceDelete},
 
+		gvr("apiextensions.k8s.io", "v1beta1", "customresourcedefinitions"): {"delete": testCRDDelete},
```

Comment:
I had to add a specific delete test for CRD, because like namespaces, apiserver does not delete CRD on an update that removes the last pending finalizer.

Reply:
should it? that seems like the same bug as the one you fixed in storage

Reply:
CRD storage doesn't override Update, and only changes Delete behavior on first delete... I don't think it should require a custom test method (I'd expect removing finalizers on update of a CRD with a deletiontimestamp set to actually delete)

Reply:
I removed the commit that modified `shouldDeleteDuringUpdate`. The function returns `false` unless the `DeletionGracePeriodSeconds` was set on the first Delete. CRD [storage](https://github.com/kubernetes/kubernetes/blob/4d3d153210c0788b8e188c467d31061470d8c042/staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresourcedefinition/etcd.go#L70) doesn't set the `DeletionGracePeriodSeconds`.

I removed my previous fix to `shouldDeleteDuringUpdate` because that would make  `Namespaces` misbehave.

I tracked the issue in https://github.com/kubernetes/kubernetes/issues/77944. We can solve it in a followup.